PF/assignment2/prg1.java

```diff
@@ -0,0 +1,81 @@
+package assignment2;
+
+public class prg1 {
+	public static void main (String args[])
+	{
+		prg1 obj= new prg1();
+		
+		int i;
+		for(i=1;i<10;i++)
+		{
+			String a="";
+			a= obj.pyramid(i,9);
+			
+			System.out.println(a);
+		}
+	}
+	
+	public String spaces(int row,int n)
+	{
+	String a= "";
+	int i;
+				if(row<(n+1)/2)
+				{
+			for(i=5;i>row;i--)
+			{
+				a= a + " "+ " ";
+			}
+				}
+				else
+				{
+					for(i=1;i<(row-4);i++)
+					{
+						
+							a= a + " "+ " ";
+						
+					}
+				}
+		
+		return a;		
+	}
+
+	public String number(int row, int n)
+	{
+		String a= "";
+		int i,j;
+				int b= n-row +1;	
```

Comment:
ACTION: Try to optimize code. It can done like that

int i,j,b;
		if(row<(n+1)/2)
		{
			b = row;
		}
		else
		{
			b = n- row +1;
		}
		
		for(i=1;i<b;i++)
		{
			a= a+" "  + i;
		}
		
		for(j=b;j>0;j--)
		{
			a= a+" " + j;
		}